tests/phpunit/tests/user/author.php

```diff
@@ -144,4 +145,42 @@ public function test_get_the_author_posts_link_with_permalinks() {
 		unset( $GLOBALS['authordata'] );
 	}
 
+	/**
+	 * @ticket 51859
+	 */
+	public function test_get_author_url_link() {
+		$author_url          = get_the_author_meta( 'url' );
+		$author_display_name = get_the_author();
+
+		$author_url_link_html = sprintf(
+			'<a href="%1$s" title="%2$s" rel="author external">%3$s</a>',
+			esc_url( $author_url ),
+			esc_attr( sprintf( 'Visit %s&#8217;s website', $author_display_name ) ),
+			$author_display_name
+		);
+
+		$this->assertSame( get_the_author_link(), $author_url_link_html );
+	}
+
+	public function filter_author_url_link_html( $link, $url, $author_data ) {
+		return '<a href="' . $url . '">' . $author_data->display_name . '</a>';
```

Comment:
If you decide to use the filter rather than the mock, then I suggest putting `data-ticket='51859'` in as an attribute to make sure it's your expected value.